python/ql/src/experimental/Security/XmlRpcDottedNames.ql

```diff
@@ -0,0 +1,24 @@
+/**
+ * @name The allow_dotted_names option may allow intruders to execute arbitrary code
+ * @description Enabling the allow_dotted_names option allows intruders to access your module’s global variables and may allow intruders to execute arbitrary code on your machine.
+ * @kind problem
+ * @problem.severity warning
+ * @id python/xml-rpc-dotted-names
+ * @tags reliability
+ *       security
+ */
+
+import python
+
+from CallNode call, ControlFlowNode allow_dotted_names, Attribute a
+where
+  a.getLocation().getStartLine() = call.getLocation().getStartLine() and
+  a.getName() = "register_instance" and
+  not call.getLocation().getFile().inStdlib() and
+  (
+    allow_dotted_names = call.getArgByName("allow_dotted_names") or
+    allow_dotted_names = call.getArg(1)
+  ) and
+  allow_dotted_names.getNode().toString() = "True"
+select a,
+  "Enabling the allow_dotted_names option allows intruders to access your module’s global variables and may allow intruders to execute arbitrary code on your machine."
```

Comment:
A modern way to write this would be
```suggestion
import python
import semmle.python.ApiGraphs

from API::CallNode call, DataFlow::Node allow_dotted_names
where
  call =
    API::moduleImport("xmlrpc")
        .getMember("server")
        .getMember("SimpleXMLRPCServer")
        .getReturn()
        .getMember("register_instance")
        .getACall() and
  allow_dotted_names = call.getParameter(1, "allow_dotted_names").getAValueReachingSink() and
  allow_dotted_names.asExpr() instanceof True
select allow_dotted_names,
  "Enabling the allow_dotted_names option allows intruders to access your module’s global variables and may allow intruders to execute arbitrary code on your machine."

```